include/game.hpp

```diff
@@ -0,0 +1,116 @@
+#ifndef GAME_HPP_
+#define GAME_HPP_
+
+#include <json_fwd.hpp>
+#include <map>
+#include <memory>
+#include <utility>
+#include <vector>
+#include "card_research.hpp"
+#include "character.hpp"
+#include "map.hpp"
+#include "runebound_fwd.hpp"
+#include "tokens.hpp"
+
+namespace runebound {
+const int DECK_SIZE = 60;
+
+namespace game {
+void to_json(nlohmann::json &json, const Game &game);
+void from_json(const nlohmann::json &json, Game &game);
+
+struct Game {
+private:
+    ::runebound::map::Map m_map;
+    std::vector<::runebound::character::Character> m_characters;
+    std::vector<unsigned int> m_card_deck;
+    std::map<::runebound::token::Token, unsigned int> m_tokens;
+    unsigned int m_turn = 0;
+    unsigned int m_count_players;
+    std::vector<unsigned int> m_indexes_card_research;
+    const std::vector<cards::CardAdventure *> ALL_CARDS;
+
+    std::vector<cards::CardAdventure *> generate_all_cards();
+
+    template <typename T>
+    void pop_element_from_vector(T element, std::vector<T> &vec) {
+        for (std::size_t i = 0; i < vec.size(); ++i) {

```

Comment:
that should be `erase-remove` idiom
https://en.wikipedia.org/wiki/Erase%E2%80%93remove_idiom

tldr
```
vec.erase(std::remove(vec.begin(), vec.end(), element), v.end());
```